From 92ea7d19e6f7cb93a1b00eb63ed6d295ada60936 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Tue, 12 Aug 2014 18:28:36 +0800
Subject: [PATCH 4/5] ENGR00324853 Update for l-preview build

convert offset to unsigned value, or mmap for
offset > 0x80000000 will fail.

Change-Id: I1a7de9cb4f5c10310c6afe23e648e42e7757d5e7
Signed-off-by: Fang Hui <b31070@freescale.com>
---
 libc/bionic/mmap.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libc/bionic/mmap.cpp b/libc/bionic/mmap.cpp
index 8f25a89..5077cfa 100644
--- a/libc/bionic/mmap.cpp
+++ b/libc/bionic/mmap.cpp
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -60,5 +61,6 @@ void* mmap64(void* addr, size_t size, int prot, int flags, int fd, off64_t offse
 }
 
 void* mmap(void* addr, size_t size, int prot, int flags, int fd, off_t offset) {
-  return mmap64(addr, size, prot, flags, fd, static_cast<off64_t>(offset));
+  size_t unsigned_offset = static_cast<size_t>(offset); // To avoid sign extension.
+  return mmap64(addr, size, prot, flags, fd, static_cast<off64_t>(unsigned_offset));
 }
-- 
1.8.0

